salvo/src/lib/source_tree.py

```diff
@@ -108,15 +128,40 @@ def get_origin(self) -> str:
         HEAD and the local HEAD.  This diff is applied to the source in the
         remote context.
     """
-    return ''
-
-  def get_directory(self) -> str:
+    valid = self._validate()
+    log.debug(f"Valid: {valid} for object: {self}")
+
+    origin_url = self._source_repo.source_url
+    output_directory = self.get_source_directory()
+
+    cmd = "git remote -v"
+    if not origin_url:
+      cmd_params = cmd_exec.CommandParameters(cwd=output_directory)
+      output = cmd_exec.run_command(cmd, cmd_params)
+      for line in output.split('\n'):
+        match = re.match(_GIT_ORIGIN_REGEX, line)
+        if match:
+          origin_url = match.group(1)
+          self._source_repo.source_url = origin_url
+          break
+
+    if not origin_url:
+      raise SourceTreeError(
```

Comment:
Can we add the `Raises:` section into the docstring now that the method can raise an exception?

Reply:
Certainly.  Updated.